test/L2GovernanceRelay.t.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+// Copyright (C) 2024 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.21;
+
+import "dss-test/DssTest.sol";
+
+import { L2GovernanceRelay } from "src/L2GovernanceRelay.sol";
+import { MessengerMock } from "test/mocks/MessengerMock.sol";
+
+contract L2SpellMock {
+    function exec() external {}
+    function revt() pure external { revert("L2SpellMock/revt"); }
+}
+
+contract L2GovernanceRelayTest is DssTest {
+
+    L2GovernanceRelay relay;
+    address l1GovRelay = address(0x111);
+    MessengerMock messenger;
+    address spell;
+
+    function setUp() public {
+        messenger = new MessengerMock();
+        messenger.setXDomainMessageSender(l1GovRelay);
```

Comment:
It seems unnecessary.

Reply:
This was meant to make it more clear that the first revert occurs due to a wrong msg.sender and the next revert occurs due to a wrong xDomainMessageSender. Have rearranged the test slightly to make it more clear.

Reply:
Yeah imagined it was related to that. I was thinking we could have also checked the xDomainMessageSender part first and then the msg.sender one. That could remove the need of the two extra setXDomainMessageSender, but maybe this way is better as being more explicit, so resolving this.